### PR TITLE
add httpsign-cli

### PR DESCRIPTION
**Pull request by favadi**

### CODE REVIEW

File: authenticator/authenticator.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package authenticator
+
+import (
+	"github.com/gin-contrib/httpsign"
+	"github.com/gin-contrib/httpsign/crypto"
+	"github.com/gin-contrib/httpsign/validator"
+)
+
+// NewAuthenticator create a httpsign.Authenticator to check the message signing is valid or not
+func NewAuthenticator(readKeyID, readKeySecret, writeKeyID, writeKeySecret string) (*httpsign.Authenticator, error) {
```
Comment (NNagato) [inline authenticator/authenticator.go]: There are no error in this function, why do we return error here?

File: authenticator/nonce_validator.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+// NonceValidator checking validate by time range
+type NonceValidator struct {
+	// TimeGap is max time different between client submit timestamp
+	// and server time that considered valid. The time precision is millisecond.
+	TimeGap uint64
+}
+
+// NewNonceValidator return NonceValidator with default value (30 second)
+func NewNonceValidator() *NonceValidator {
```
Comment (NNagato) [inline authenticator/nonce_validator.go]: I think this function should have param to modify TimeGap

File: authenticator/authenticator.go (hunk 1 above)
Comment (favadi) [reply authenticator/authenticator.go]: fixed in 22dc64dfa78f517fcf47e19d433334d4e6e410f4.

File: authenticator/nonce_validator.go (hunk 2 above)
Comment (favadi) [reply authenticator/nonce_validator.go]: > I think this function should have param to modify TimeGap

did that in ff05ebf1faab1d43286f9cabed7e9ac5547723ea.

File: authenticator/nonce_validator.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// NonceValidatorWithMaxTimeGap is the option to create NonceValidator with custom time gap.
+func NonceValidatorWithMaxTimeGap(gap time.Duration) NonceValidatorOption {
+	return func(validator *NonceValidator) {
+		validator.MaxTimeGap = gap
+	}
+}
+
+// NewNonceValidator return NonceValidator with default value (30 second)
+func NewNonceValidator(options ...NonceValidatorOption) *NonceValidator {
+	v := &NonceValidator{
```
Comment (NNagato) [inline authenticator/nonce_validator.go:35]: Can we just define object with default value like v := &NonceValidator{}? cause in line 41 if an NonceValidatorOption change MaxTimeGap to zero we have to assign MaxTimeGap = defaultMaxTimeGap twice.

Comment (favadi) [reply authenticator/nonce_validator.go:35]: yes, we should, good spot.

Comment (favadi) [reply authenticator/nonce_validator.go:35]: did that in 724a846c3429c0697e8f92110f71aa5f226b88dc.

**Review by NNagato — APPROVED** [review]

**Pull request merged**